tests/test_snowflake_trace_id_generator.py

```diff
@@ -0,0 +1,43 @@
+import unittest
+from unittest.mock import patch
+
+from opentelemetry import trace
+from snowflake.telemetry.trace import SnowflakeTraceIdGenerator
+
+MOCK_TIMESTAMP = 1719588243.3379807
+INVALID_TRACE_ID = 0x00000000000000000000000000000000
+TRACE_ID_MAX_VALUE = 2**128 - 1
+SPAN_ID_MAX_VALUE = 2**64 - 1
+
+
+class TestSnowflakeTraceIdGenerator(unittest.TestCase):
+
+    def test_valid_span_id(self):
+        id_generator = SnowflakeTraceIdGenerator()
+        self.assertTrue(trace.INVALID_SPAN_ID < id_generator.generate_span_id() <= SPAN_ID_MAX_VALUE)
+        self.assertTrue(trace.INVALID_SPAN_ID < id_generator.generate_span_id() <= SPAN_ID_MAX_VALUE)
+        self.assertTrue(trace.INVALID_SPAN_ID < id_generator.generate_span_id() <= SPAN_ID_MAX_VALUE)
+        self.assertTrue(trace.INVALID_SPAN_ID < id_generator.generate_span_id() <= SPAN_ID_MAX_VALUE)
+
+
+    @patch('time.time', return_value=MOCK_TIMESTAMP)
```

Comment:
wow, seems like we can mock direct time.time in python? wish we had something similar in Java (where we have to only mock clock) and C++ (I don't know any way to mock time).